### PR TITLE
Fix selected list to display in dropdown

### LINKED ISSUE #69: Bug: selected list does not display in dropdown
Image

### PR DESCRIPTION
**Pull request by TishShaw**
## Description

This pull request displays the selected list name in the dropdown once selected by a user.

## Related Issue
#69 

## Acceptance Criteria
Selected list name is displayed once clicked in dropdown.

## Type of Changes
Bug fix

### Before
Screenshot 2024-03-23 at 5 44 31 AM

### After
Screenshot 2024-03-23 at 5 37 25 AM

## Testing Steps / QA Criteria

1. The user has a list of shopping lists.
2. Once the user selects a specific list in the dropdown it is shown to be selected in the dropdown and matches the list name on the top on the web page.

### CODE REVIEW

File: src/context/UserProvider.jsx [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { UserContext } from './user';
 export function UserProvider({ children }) {
 	const [user, setUser] = useState(null);
 
+	console.log(user);
```
Comment (jtkabenni) [inline src/context/UserProvider.jsx]: Remove

File: src/components/SelectListForm.jsx [hunk 2]
```diff
@@ -36,11 +40,6 @@ export default function SelectListForm({
 					<button onClick={() => navigate('/list')}>View List</button>
 				</div>
 			)}
-			<div>
-				<h3>
```
Comment (jtkabenni) [inline src/components/SelectListForm.jsx]: I think we want to keep this h3 inside the div and where it is originally. this is according to the lofi mockup layout.

**Review by jtkabenni — CHANGES_REQUESTED** [review]
Hey Tish, right now if I select a list and then refresh, the list name dissapears from the dropdown. The same thing happens if I go to the list view, then return to the home page. 
Screenshot 2024-03-29 at 12 18 41 AM

File: src/components/SelectListForm.jsx
Comment (aberger3647) [inline src/components/SelectListForm.jsx:1]: We aren't using listPath that was sent in as props from Home. What if we sent in listName instead and used listName as the initial value of the useState for setSelectedList?

If !selectedList, we can set it to 'Select a list' so it's not blank

**Review by aberger3647 — CHANGES_REQUESTED** [review]

File: src/components/SelectListForm.jsx
Comment (jtkabenni) [reply src/components/SelectListForm.jsx:1]: Yes that's a good idea!! @aberger3647

File: src/components/SelectListForm.jsx [hunk 3]
```diff
@@ -3,17 +3,19 @@ import { useState } from 'react';
 
 export default function SelectListForm({
 	data,
-	listPath,
 	areListsLoading,
+	listName,
```
Comment (jtkabenni) [inline src/components/SelectListForm.jsx]: You need to make sure this prop is actually being passed in. Where is it being passed in from?

File: src/components/SelectListForm.jsx [hunk 4]
```diff
@@ -3,17 +3,19 @@ import { useState } from 'react';
...
 	setListPath,
 }) {
-	const [selectedList, setSelectedList] = useState('');
+	const [selectedList, setSelectedList] = useState(listName);
```
Comment (jtkabenni) [inline src/components/SelectListForm.jsx]: You can set a default state here for if no list is selected, something like "No list Selected"

**Review by jtkabenni — CHANGES_REQUESTED** [review]
Hey Tish, still not seeing the expected behavior. I would make sure whatever is needed to display the listName in the select dropdown is actually getting passed to the correct component. 
Screenshot 2024-03-30 at 11 44 16 AM



Also, when there is no list selected, it is showing a blank dropdown when it should display something like "Select a list", as Alex mentioned in her comment. You can take a look at her comment to see a suggested way of implementing this. 

Screenshot 2024-03-30 at 11 48 20 AM


I would also recommend running through the user flow locally to make sure the expected behavior exists before re-requesting reviews. Thanks!

File: src/components/SelectListForm.jsx (hunk 4 above)
Comment (aberger3647) [reply src/components/SelectListForm.jsx]: @jtkabenni I tried this and it worked while on the Home page, but when I viewed the list and returned to the home page, it went back to that default state. The only way I have been able to make this work the way we expect is to set the default to listName, and add a condition in the component body:
`	if (!listName) {
		setSelectedList('No list selected')
	}`

**Comment by aberger3647** [comment]
Currently, if I select a list and then go to that list, then return to the home page, it still shows "Select a list". See my reply to Jenny for a suggestion on fixing that behavior

**Review by aberger3647 — APPROVED** [review]
Great solution!

**Review by jtkabenni — APPROVED** [review]

**Pull request merged**